StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSavedPaymentMethodsViewController.swift

```diff
@@ -435,6 +436,20 @@ class CustomerSavedPaymentMethodsViewController: UIViewController {
                             self.csCompletion?(.selected(paymentOptionSelection))
                         }
                     }
+                    // if the sync default feature is enabled, set the selected payment method as default
+                    if paymentMethodSyncDefault, let defaultPaymentMethod = selectedPaymentOption.savedPaymentMethod, let customerId = defaultPaymentMethod.customerId {
+                        Task {
+                            do {
+                                _ = try await self.customerSheetDataSource.setAsDefaultPaymentMethod(paymentMethodId: defaultPaymentMethod.stripeId, customerID: customerId)
+                                STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: .customerSheetUpdateCard)
+                            } catch {
+                                self.error = error
+                                let errorAnalytic = ErrorAnalytic(event: .customerSheetUpdateCardFailed,
+                                                                  error: Error.updatePaymentMethodFailed)
+                                STPAnalyticsClient.sharedClient.log(analytic: errorAnalytic)
+                            }
+                        }
+                    }
```

Comment:
I believe this work should be under the work that is happening somewhere under `setSelectablePaymentMethodAnimateButton` because handles `processingInFlight`'s state and updates the UI.